utility.ml

```diff
@@ -918,3 +918,11 @@ let string_of_float' : float -> string =
 
 let time_seconds() = int_of_float (Unix.time())
 let time_milliseconds() = int_of_float (Unix.gettimeofday() *. 1000.0)
+
+let queue_as_list q =
```

Comment:
I suggest extending the `Queue` module with functions `to_list` and `of_list`, i.e.
```ocaml
module Queue = struct
  include Queue
  let to_list q = ....
  let of_list xs = ...
end
```